### PR TITLE
Update release README.md

### LINKED ISSUE #1850: release/README.md needs updating
The istio/release/README.md needs updating following the monorepo switcharoo. It has some broken links and some other outdated content.

### PR DESCRIPTION
**Pull request by hklai**
**What this PR does / why we need it**:
The release README.md is out of date and this updates it.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #1850

**Special notes for your reviewer**:
Feel free to suggest change, clarification, or addition.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: release/README.md [hunk 1]
```diff
@@ -2,253 +2,31 @@
...
-The release is started from the [istio/istio](https://github.com/istio/istio) module.
+Starting [0.3.0](https://github.com/istio/istio/releases/tag/0.3.0), Istio is released and published every month and the
+monthly releases can all be found on https://github.com/istio/istio/releases. Each monthy release has its own minor version
+number, while the patch version indicates important bug fixes in the corresponding monthly release. You can find more
+information about version semantic, cadence, and support in [Istio Release Principles](https://goo.gl/dcSBxF).
 
-Istio release is currently composed of artifacts for the following repos:
+Internally, Istio releases are cut, tested, and qualified every day. And once a week, a daily release will go through
+stability and performance testing to detect regression. The monthly release mentioned above is promoted from a weekly release
+that passed all fumnctional, stability, and performance testing. You can find more information about this process in
```
Comment (mattdelco) [inline release/README.md]: typo on "fumnctional"

File: release/README.md [hunk 2]
```diff
@@ -2,253 +2,31 @@
...
-    + [Rebuild artifacts to include the tags](#rebuild-artifacts-to-include-the-tags)
-    + [Updating ```istio.VERSION```](#updating----istioversion---)
-    + [Creating archives](#creating-archives)
-    + [Finalizing the release](#finalizing-the-release)
+  * [Request Emergency Bug Fix and Patch Release](#request-emergency-bug-fix-and-patch-release) 
 
 ## Overview
 
-The release is started from the [istio/istio](https://github.com/istio/istio) module.
+Starting [0.3.0](https://github.com/istio/istio/releases/tag/0.3.0), Istio is released and published every month and the
```
Comment (mattdelco) [inline release/README.md]: I'm add a "with" after "Starting"

File: release/README.md [hunk 3]
```diff
@@ -2,253 +2,31 @@
...
+monthly releases can all be found on https://github.com/istio/istio/releases. Each monthy release has its own minor version
```
Comment (mattdelco) [inline release/README.md]: Missing an L in "Each monthy release"

File: release/README.md [hunk 4]
```diff
@@ -2,253 +2,31 @@
...
-and edit the release that points to ```${RELEASE_TAG}```. Uploads the artifacts created by the previous script.
-
-
-### Finalizing the release
-
-Create a PR, where you increment ```istio.RELEASE``` for the next
-release and you update ```istio/downloadIstio.sh``` to point to ```${RELEASE_TAG}```
+When a severe bug is found, developers can contact hklai@google.com and vickyxu@google.com to create a patch release on all
+[currently supported releases]
+(https://docs.google.com/document/d/1scC_f7nObwBut_xxE6lUfvcHBybmttUsJDMwXD3gcO4/edit#heading=h.b56ea7aggz9t). Only bugs that
```
Comment (mattdelco) [inline release/README.md]: Perhaps use the short URL https://goo.gl/dcSBxF   .  When I tried visting the full version with the #heading=... I got an error from google docs.

File: release/README.md [hunk 5]
```diff
@@ -2,253 +2,31 @@
...
-
-```
-$ TAR=gtar CP=gcp githubctl --token_file=<github token file> \
-    --op=uploadArtifacts --base_branch=<release branch or master> \
-    --next_release=<next release> --ref_sha=${RELEASE_SHA}
-```
-
-You can finalize release in github release [page](https://github.com/istio/istio/releases) before this PR got merged, but it's your responsibility to make sure it got merged for next release. **You should finalize it as an actual release (not shown as "draft") before Step 6**
- 
-### Step 6: Generating release note.
```
Comment (mattdelco) [inline release/README.md]: The release note step hasn't been automated yet so it's still a manual step.  As of yesterday I did complete an initial port of the functionality to bash though I'm not sure if we want to start using it at this phase yet.  The tool might just be "best effort" and if there's any issues or problems (or someone wants to customize the range of time to query for the notes) then someone might want to run the tool instead.

**Comment by yutongz** [comment]
Do we still want to support 0.2 patching, we get some resources(clusters, tools, docs) supporting that. If we decided not to, that we can remove them all :)

File: release/README.md (hunk 5 above)
Comment (hklai) [reply release/README.md]: The goal is to reduce the chance someone outside of the test/release group "might want to run the tool" for now, until the process (and probably the UI) is ready for external use.

File: release/README.md (hunk 1 above)
Comment (hklai) [reply release/README.md]: Done

File: release/README.md (hunk 3 above)
Comment (hklai) [reply release/README.md]: Done

File: release/README.md (hunk 2 above)
Comment (hklai) [reply release/README.md]: Done

File: release/README.md (hunk 4 above)
Comment (hklai) [reply release/README.md]: #heading won't work using the shorten link.

Could you try again? It seems to work for me just now.

**Comment by hklai** [comment]
I don't suppose we want to patch 0.2 anymore. Even so, the previous release step did not work AFAICT.

I intend to replace that with general information about release for now, and add more information when the process is ready for others outside of the group to run.

**Comment by ldemailly** [comment]
I am a bit concerned by the information disappearing and not being replaced?

**Comment by hklai** [comment]
@ldemailly The old scripts do not work anymore after repo merge and some of them had been deleted. There is no point to keep them around.

As mentioned in the doc, the group is solely responsible for release until the process/UI are ready for broader use (next quarter). At that time we will add more information back here.

**Comment by mattdelco** [comment]
I thought there was something on the github or istio.io websites that linked to the page (so it had a potential to be useful for someone who was tasked to do a release 0.2 patch but didn't think to look on the release-0.2 branch for the relevant version of the doc), but I didn't find it this afternoon.  If 0.2 is dead then it's a moot point.

I can see @ldemailly 's concern about removing old content but not adding in new content.  I suppose there will be content to add once the process/UI is ready for others to use. I'm not clear on whether we will also eventually add content that explains how to manually kick things off (in case of bug or service outage, or for outside contributors to dev test).  We lack turnkey information on how outside contributors can build Istio (e.g., for their own dev testing), though the old content wasn't really usable by these people either so it's not a new concern.

In summary I don't have anything new to suggest for this PR.  Afaik the system thinks I'm a nobody when it comes to approvals so there's no point in me using any keywords.

**Comment by yutongz** [comment]
@mattdelco 
Don't be sad. You are in the istio org. So you should be able to say "/lgtm" :)

**Comment by hklai** [comment]
@mattdelco Hiding those scripts behind the release process (i.e. airflow or later the release UI) is the way to go and I don't think we want others to really mess with release scripts manually. AFAICT, nobody other than us did the past releases anyway. For anyone else who really wants to create a release themselves, they can contact us for guidance. I suppose it is very rare.

As I said earlier, when airflow or release UI is ready for wider use, we will add more information here.

Also, I think you should add yourself to the OWNERS file since you are now the release king. =)
